ms/MSSel/MSSpwGramPP.cc

```diff
@@ -0,0 +1,217 @@
+//# because keyword register is not supported anymore.
+#define register
+#include "MSSpwGramPP.ycc"                  // bison output
+#include "MSSpwGramPP.lcc"                  // flex output
+
+// Define the yywrap function for flex.
+int MSSpwGramPPwrap()
+{
+  return 1;
+}
+
+namespace casacore { //# NAMESPACE CASACORE - BEGIN
+  
+  //# Declare a file global pointer to a char* for the input string.
+  static const char*           strpMSSpwGramPP = 0;
+  static Int                   posMSSpwGramPP = 0;
+  static string tt="";
+  // MSSpwGramwrap out of namespace
+  
+  string& getTT() {return tt;}
+  //# Parse the command.
+  //# Do a yyrestart(yyin) first to make the flex scanner reentrant.
+  int msSpwGramPPParseCommand (const MeasurementSet* ms, const String& command) 
+  {
+    try 
+      {
+	Int ret;
+	MSSpwGramPPrestart (MSSpwGramPPin);
+	yy_start = 1;
+	strpMSSpwGramPP = command.chars();     // get pointer to command string
+	posMSSpwGramPP  = 0;                   // initialize string position
+	MSSpwParse parser(ms);               // setup measurement set
+	MSSpwParse::thisMSSParser = &parser; // The global pointer to the parser
+	parser.reset();
+	ret=MSSpwGramPPparse();                // parse command string
+	return ret;
+      }
+    catch (MSSelectionSpwError &x)
+      {
+	String newMesgs;
+	newMesgs = constructMessage(msSpwGramPPPosition(), command);
+	x.addMessage(newMesgs);
+	throw;
+      }
+  }
+  
+  int baseMSSpwGramPPParseCommand (MSSpwParse* parser, const String& command,
+				 Vector<Int>& selectedIDs, Matrix<Int>&selectedChans,
+				 Vector<Int>& selectedDDIDs) 
+  {
+    try 
+      {
+	Int ret;
+	MSSpwGramPPrestart (MSSpwGramPPin);
+	yy_start = 1;
+	strpMSSpwGramPP = command.chars();     // get pointer to command string
+	posMSSpwGramPP  = 0;                   // initialize string position
+	MSSpwParse::thisMSSParser = parser; // The global pointer to the parser
+	parser->reset();
+	ret=MSSpwGramPPparse();                // parse command string
+	selectedIDs = parser->selectedIDs();
+	selectedChans = parser->selectedChanIDs();
+	selectedDDIDs = parser->selectedDDIDs();
+	if ((selectedIDs.size() == 0) ||
+	    (selectedChans.size() == 0))
+	  throw(MSSelectionSpwParseError("No valid SPW & Chan combination found"));
+	return ret;
+      }
+    catch (MSSelectionSpwError &x)
+      {
+	String newMesgs;
+	newMesgs = constructMessage(msSpwGramPPPosition(), command);
+	x.addMessage(newMesgs);
+	throw;
+      }
+  }
+  
+  int msSpwGramPPParseCommand (const MSSpectralWindow& spwSubTable, 
+			     const MSDataDescription& ddSubTable, 
+			     const TableExprNode& colAsTEN,
+			     const String& command,
+			     Vector<Int>& selectedIDs,
+			     Matrix<Int>& selectedChans,
+			     Vector<Int>& selectedDDIDs) 
+  {
+    Int ret;
+    //    MSSpwParse *thisParser = new MSSpwParse(ms);
+    MSSpwParse *thisParser = new MSSpwParse(spwSubTable, ddSubTable, colAsTEN);
```

Comment:
Why is this object allocated with new and not simply on the stack like
     MSSpwParse thisParser(spwSubTable, ddSubTable, colAsTEN);
which removes the need of destroying the object and catching the exception?